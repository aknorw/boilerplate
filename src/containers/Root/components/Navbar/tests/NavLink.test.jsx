import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import NavLink from '../NavLink'

console.error = jest.fn()

describe('<NavLink />', () => {
  it('should be defined', () => {
    expect(NavLink).toBeDefined()
  })
  it('should log 1 error when `to` is missing', () => {
    shallow(<NavLink />)
    expect(console.error).toHaveBeenCalledTimes(1)
  })
  it('should render correctly', () => {
    const tree = shallow(<NavLink to="/test" />)
    expect(tree).toMatchSnapshot()
  })
})
