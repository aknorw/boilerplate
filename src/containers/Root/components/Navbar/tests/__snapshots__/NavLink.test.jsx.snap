// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLink /> should render correctly 1`] = `
.c0 {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  -webkit-font-smoothing: antialiased;
}

<Link
  className="c0"
  replace={false}
  to="/test"
/>
`;
